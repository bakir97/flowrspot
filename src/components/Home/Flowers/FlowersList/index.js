import FlowersList from "./FlowersList";
export default FlowersList;
