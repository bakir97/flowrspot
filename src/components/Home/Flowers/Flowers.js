import React, { Component } from "react";
import FlowersList from "./FlowersList";
import styles from "./Flowers.module.scss";
import Loader from "../../../sharedComponents/Loader";
export default class Flowers extends Component {
  componentDidMount() {
    this.props.getFlowers();
  }
  render() {
    const { flowers, isLoading } = this.props;
    return isLoading ? <Loader /> : <FlowersList flowers={flowers} />;
  }
}
