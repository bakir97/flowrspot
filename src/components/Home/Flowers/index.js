import Flowers from "./Flowers";
import { connect } from "react-redux";
import { getFlowers, addFavFlower } from "../../../redux/flowers/actions";
import { getFlowersState, isLoading } from "../../../redux/flowers/selectors";
import { getUser } from "../../../redux/user/selectors";
const mapStateToProps = state => ({
  flowers: getFlowersState(state),
  isLoading: isLoading(state),
  user: getUser(state)
});

const mapDispatchToProps = { getFlowers, addFavFlower };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Flowers);
