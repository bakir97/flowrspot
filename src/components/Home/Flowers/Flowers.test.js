import React from "react";
import { shallow } from "enzyme";
import Flowers from "./Flowers";

describe("Flowers", () => {
  it("should render correctly", () => {
    const mockFn = jest.fn();
    const component = shallow(<Flowers getFlowers={mockFn} />);
    expect(component).toMatchSnapshot();
  });
  it("should render FlowersList", () => {
    const mockFn = jest.fn();

    const component = shallow(<Flowers getFlowers={mockFn} />);
    expect(component).toMatchSnapshot();
  });
  it("should render Loader", () => {
    const mockFn = jest.fn();

    const component = shallow(<Flowers isLoading getFlowers={mockFn} />);
    expect(component).toMatchSnapshot();
  });
  it("should call getFlowers", () => {
    const mockFn = jest.fn();
    shallow(<Flowers getFlowers={mockFn} />);
    expect(mockFn).toHaveBeenCalled();
  });
});
