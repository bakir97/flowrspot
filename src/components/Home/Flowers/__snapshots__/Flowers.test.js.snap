// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flowers should render FlowersList 1`] = `<FlowersList />`;

exports[`Flowers should render Loader 1`] = `<Loader />`;

exports[`Flowers should render correctly 1`] = `<FlowersList />`;
